main/pipeline/pipeline.cc

```diff
@@ -810,6 +810,173 @@ void unpartitionPackageFiles(vector<ast::ParsedFile> &packageFiles, vector<ast::
     }
 }
 
+namespace {
+
+bool isTestExport(const ast::ExpressionPtr &expr) {
+    auto send = ast::cast_tree<ast::Send>(expr);
+    if (!send || send->fun != core::Names::export_() || send->numPosArgs() != 1) {
+        return false;
+    }
+
+    auto sym = ast::cast_tree<ast::UnresolvedConstantLit>(send->getPosArg(0));
+    while (sym) {
+        if (ast::isa_tree<ast::EmptyTree>(sym->scope)) {
+            return sym->cnst == core::Names::Constants::Test();
+        }
+
+        if (auto parent = ast::cast_tree<ast::UnresolvedConstantLit>(sym->scope)) {
+            sym = parent;
+        } else {
+            break;
+        }
+    }
+
+    return false;
+}
+
+ast::ParsedFile copyPackageWithoutTestExports(const core::GlobalState &gs, const ast::ParsedFile &ast) {
```

Comment:
Should we put a comment here that we'd like to make this function redundant eventually, because test packages are their own thing, and what the motivation for needing to do this in the mean time is (e.g. resolver issues)?

Reply:
Good idea, I'll add that.